### ISSUE TITLE
add `Http.collectManaged` 

### ISSUE THREAD

**Issue by tusharmath** [zio/zio-http#866]
**Describe the solution you'd like**
This should be possible
```scala
Http.collectManaged[Request] {
 case GET -> !! / "users" => ZManaged.succeed(Response.ok)
}
```

**Comment by beem812** [comment]
I think I may be able to take this one, I have something that appears to work but I'm fairly new to the ZIO world. If I can take this one my first question is going to be: Is it appropriate to call `useNow` on the ZManaged to get a ZIO here?

**Comment by beem812** [comment]
I forgot to link to this issue with my PR but I believe this is closed with #909

**Comment by girdharshubham** [comment]
Thanks for this contribution @beem812. Also, congratulations on your first contribution.
